### DISCUSSION TITLE
🎉 2.0.0

### DISCUSSION THREAD

**Discussion by giorgiabosello** [giorgiabosello/google-maps-react-markers#109] [Announcements]
# 💥 Breaking Changes
## Library has completely been rewritten in Typescript!
🚧 Builded with [tsup](https://github.com/egoist/tsup)
✏️ Fully typed
🔥 PropTypes removed
⬆️ Dependencies upgrade (see below)

### What's Changed
* ⬆️ Bump gh-pages from 5.0.0 to 6.0.0 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/93
* ⬆️ Bump @babel/preset-env from 7.22.10 to 7.22.14 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/105
* ⬆️ Bump typescript from 5.1.6 to 5.2.2 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/106
* ⬆️ Bump prettier from 3.0.2 to 3.0.3 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/107
* ⬆️ Bump eslint-plugin-n from 16.0.1 to 16.0.2 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/108
* ⬆️ Bump @babel/eslint-parser from 7.22.10 to 7.22.11 by @dependabot in https://github.com/giorgiabosello/google-maps-react-markers/pull/104

**Full Changelog**: https://github.com/giorgiabosello/google-maps-react-markers/compare/1.4.2...2.0.0-alpha.0

<hr /><em>This discussion was created from the release <a href='https://github.com/giorgiabosello/google-maps-react-markers/releases/tag/2.0.0-alpha.0'>2.0.0-alpha.0</a>.</em>

<hr/>

🙏🏼 Since I'm new to Typescript, I would really need some feedback. Thank you!

**Comment by Zak-JS** [comment]
Just wanted to say good work!

**Reply by giorgiabosello** [reply]
Thank you!

**Reply by melloware** [reply]
@giorgiabosello absolutely fantastic work and we all appreciate keeping this library up to date and I am very happy its with Typescript!

**Comment by leomar-developer** [comment]
Is there a way to reference all the types defined within this package? I plan to use this library but I need to build some types on top of the ones used internally by this package, tried something like:

`import { MapProps } from 'google-maps-react-markers'`

But did not work, is there a separate package with the types?

**Reply by giorgiabosello** [reply]
@leomar-developer nope, not yet. I forgot to export them. Give me a few hours and I'll update the repository and get back to you.
Thanks

**Reply by leomar-developer** [reply]
Thank you @giorgiabosello ! looking forward to it

**Reply by giorgiabosello** [reply]
@leomar-developer released in [version 2.0.1](https://github.com/giorgiabosello/google-maps-react-markers/releases/tag/2.0.1). Let me know if everything's right :)

**Reply by leomar-developer** [reply]
@giorgiabosello exactly what I needed, thanks a lot!

**Comment by parvejkhan-seraphic** [comment]
i want to show popup and infowindow when i hover and click on marker how can i achieve this

**Reply by giorgiabosello** [reply]
@parvejkhan-seraphic You need to manage them by yourself in your Marker component. Props are forwarded :)

**Comment by yoyoyo8x** [comment]
Thank you for the wonderful work. Does this package support localization? I want to change it to Korean.

**Reply by giorgiabosello** [reply]
@yoyoyo8x yes, just pass in language props in `externalApiParams`. eg.:
```ts
<GoogleMap
  ...
  externalApiParams={{ region: 'JP', language: 'ja' }}
/>
```

**Reply by yoyoyo8x** [reply]
Thanks a lot!

**Comment by sbayeganeh** [comment]
How do I get coordinates of a point when I click on the map but not a Marker?

**Reply by giorgiabosello** [reply]
See [here](https://github.com/giorgiabosello/google-maps-react-markers/discussions/101)

**Comment by andrikash** [comment]
thank you very much, you're amazing!

**Comment by shellbryson** [comment]
This has been super useful on one project. It would be amazing if support for mapIDs was rolled into this. Custom styling requires a mapId in addition to the API key. I've been trying to figure out how to make that work with little luck.

**Reply by giorgiabosello** [reply]
Hi @shellbryson do you mean on the Google Maps API script? If I forward the mapId, where do you need it then?

**Reply by shellbryson** [reply]
You can pass an ID for a particular map when using Google's API directly. This generates a _styled_ map:

```
function initMap() {
  new google.maps.Map(document.getElementById("map"), {
    mapId: "8e0a97af9386fef",
    center: { lat: 48.85, lng: 2.35 },
    zoom: 12,
  });
}
```

**Reply by giorgiabosello** [reply]
@shellbryson gotcha! Google Maps options are directly forwarded to the Map instance with the prop `options`. 
E.g.
```js
<GoogleMap
  {...props}
  options={{ mapId: "8e0a97af9386fef" }}
/>
```
You can see the code [here](https://github.com/giorgiabosello/google-maps-react-markers/blob/master/src/map/map.tsx#L72). :)

**Reply by shellbryson** [reply]
So it does. That works perfectly for me, and I really should've spotted that 😂 Perhaps more coffee next time... Thank you!
